src/Infrastructure.EntityFramework/Repositories/TaxRateRepository.cs

```diff
@@ -17,9 +19,9 @@ public async Task ArchiveAsync(Core.Entities.TaxRate model)
         using (var scope = ServiceScopeFactory.CreateScope())
         {
             var dbContext = GetDatabaseContext(scope);
-            var entity = await dbContext.FindAsync<Core.Entities.TaxRate>(model);
-            entity.Active = false;
-            await dbContext.SaveChangesAsync();
+            await dbContext.TaxRates
+                .Where(tr => tr.Id == model.Id)
+                .ExecuteUpdateAsync(property => property.SetProperty(tr => tr.Active, false));
```

Comment:
This method would have completely failed before. `model` isn't the key for `TaxRate`, we should have been passing in `model.Id`. If I just fixed that it would have thrown a `NullReferenceException` when calling `entity.Active` and the given tax rate doesn't exist. This new code behaves much more like the [SQL Server one](https://github.com/bitwarden/server/blob/cca358c88caacfbfbb8030580a567dc8d64ee5a1/src/Sql/dbo/Stored%20Procedures/TaxRate_Archive.sql#L7-L12). It doesn't check at all if a tax rate was actually archived.

...also I don't believe this table gets used at all in SH so this doesn't really matter.